File: CSharp/models/Cordyceps.cs
namespace models
{
    public abstract class Cordyceps
    {
        // atributo privado (no nulo)
        private string origin = "";

        // propiedad publica 
        public string Origin
        {
            get { return origin; }
            set
            {
                if (!string.IsNullOrEmpty(value))
                    origin = value;
            }
        }

        // constructor
        public Cordyceps(string origin)
        {
            Origin = origin;
        }

        // metodo sin retorno
        public void Spread()
        {
            Console.WriteLine("El hongo se propaga desde: " + Origin);
        }

        //metodo polimorfico
        public abstract string Behavior();
    }
}

File: Java/models/Cordyceps.java
package models;

//abstraccion
public abstract class Cordyceps {
    //encapsulamiento con atributo privado
    private String origin;

    //constructor
    public Cordyceps(String origin) {
        this.origin = origin;
    }

    //metodo sin retorno
    public void spread() {
        System.out.println("El hongo se propaga desde: " + origin);
    }

    // metodo con retorno para polimorfismo
    public abstract String behavior();
}
